packages/patterns/src/types.js

```diff
@@ -514,8 +514,9 @@ export {};
  * @typedef {{
  *   klass: 'Interface',
  *   interfaceName: string,
- *   methodGuards: T
- *   sloppy?: boolean
+ *   methodGuards: { [K in keyof T]: K extends symbol ? never : T[K] },
+ *   symbolMethodGuards?: CopyMap<symbol, MethodGuard>,
```

Comment:
Nice typing of `methodGuards`. I wondered how to do that.

Could you use the same trick to type `symbolMethodGuard` with the opposing side of that conditional?

Attn @turadg

I think doing so will help @turadg 's efforts to statically inter TS types from patterns and guards.

Reply:
I got it to type the CopyMap K as a symbol-keyed entry of T, but don't know how to establish the pair relationship. It will be open for followup.